syntax = "proto3";
import "google/protobuf/empty.proto";

package fraud_detection;

service FraudDetectionService {
    rpc InitFraudDetection(FraudDetectionData) returns (google.protobuf.Empty);
    rpc FraudDetection(FraudDetectionRequest) returns (FraudDetectionResponse);
}

message User {
    string name = 1;
    string contact = 2;
}

message CreditCard {
    string number = 1;
    string expirationDate = 2;
    string cvv = 3;
}

message Transaction {
    User user = 1;
    CreditCard creditCard = 2;
    bool termsAccepted =  3;
}

message FraudDetectionData {
    string orderId = 1;
    Transaction data = 2;
}

message FraudDetectionRequest {
    string orderId = 1;
    repeated int32 vector_clock = 2;
}

message FraudDetectionResponse {
    bool is_valid = 1;
    repeated int32 vector_clock = 2;
    string message = 3;
}
